src/odemis/gui/comp/overlay/world.py

```diff
@@ -111,6 +112,170 @@ def on_left_up(self, evt):
     def draw(self, ctx, shift=(0, 0), scale=1.0):
         pass
 
+
+MODE_EDIT_FEATURES = 1
+MODE_SHOW_FEATURES = 2
+FEATURE_DIAMETER = 30  # pixels
+
+class CryoFeatureOverlay(StagePointSelectOverlay, DragMixin):
+    """ Overlay for handling showing interesting features of cryo projects """
+    def __init__(self, cnvs, features_va, tool_va=None):
+        """
+        :param cnvs: (DblMicroscopeCanvas) Canvas to which the overlay belongs
+        :param features_va: (ListVA of CryoFeature) list of the interesting cryo features in the project
+        :param tool_va: (IntEnumerated or None) the feature tool if found
+        """
+        StagePointSelectOverlay.__init__(self, cnvs)
+        DragMixin.__init__(self)
+        self._mode = MODE_SHOW_TOOLS
+
+        self._selected_tool_va = tool_va
+        if self._selected_tool_va:
+            self._selected_tool_va.subscribe(self._on_tool, init=True)
+
+        # TODO: Find a better icon
+        self._feature_icon = cairo.ImageSurface.create_from_png(guiimg.getStream('/icon/feature_icon.png').name)
+        self._icon_w_adjust = self._feature_icon.get_width() / 2
+        self._icon_h_adjust = self._feature_icon.get_height() / 2
+
+        if features_va:
+            self._features_va = features_va
+            features_va.subscribe(self._on_features_changes, init=True)
+        self._selected_feature = None
+        self._hover_feature = None
+
+    def _on_tool(self, selected_tool):
+        """ Update the feature mode (show or edit) when the overlay is active and tools change"""
+        if self.active:
+            if selected_tool == TOOL_FEATURE:
+                self._mode = MODE_EDIT_FEATURES
+            else:
+                self._mode = MODE_SHOW_FEATURES
+
+    def _on_features_changes(self, _):
+        # refresh the canvas on _features_va list change
+        self.cnvs.update_drawing()
+
+    def on_dbl_click(self, evt):
```

Comment:
The functions `on_dbl_click()`, `on_left_down()`, `on_left_up()` and `on_motion()` are callbacks right? But i do not  see connections between those functions and VA's or connections using the wx function `Bind()`. So how does this work?

Reply:
The `Overlay` abstract base class contains these bindings.